aten/src/ATen/native/cudnn/Conv.cpp

```diff
@@ -268,6 +286,61 @@ struct ConvolutionArgs {
   }
 };
 
+std::string repro_from_args(const ConvolutionArgs& args) {
```

Comment:
I'm a little ambivalent about this. On the one hand, it is cool to produce repros in this style. But on the other hand, ensuring that the sample code here doesn't go stale is not going to be too easy. A compromise might be to produce the data in question here into a generic machine readable format (e.g., JSON) and then have a little Python script on the other side that takes that data and reinterprets it into an actual sequence of torch calls.

Reply:
I'm OK with adding this in as a one off, but if we start doing this pattern in other places in our codebase, we should think about some general infrastructure for doing this sort of thing.

Reply:
We could also have this as an exposed function and test that produced python code is runnable.

Reply:
Thanks. I checked your proposal at #37160 , which is very nice and comprehensive. Since convolution is mostly guaranteed to be backward compatible and interface stable, this sample code here would work for a long time. I'm glad to refactor this in the way you mentioned when there is a better "repro code generation mechanism" available, and I'm looking forward to it.